internal/task/gitcheck/check.go

```diff
@@ -63,7 +64,23 @@ func (t Task) Run(ctx *context.Context) error {
 			out = append(out, s.String())
 		}
 
-		return ErrDirty{status: strings.Join(out, "\n")}
+		allowedFiles := ctx.Config.Git.DirtyFiles
+		if allowedFiles == nil {
+			return ErrDirty{status: strings.Join(out, "\n")}
```

Comment:
I would update the `status` field of the `ErrDirty` struct to be `[]git.StatusFile`. This allows the output from `ctx.GitClient.PorcelainStatus()` to be directly passed in. Removing the need for the call to `strings.join()`. The error can then format the output as needed. I believe the `git.StatusFile` type has implemented `String()` so you should just be able to loop over the array and print its `String()` representation without breaking anything